### PR TITLE
distask: make the test stable

### PR DESCRIPTION
**Pull request by zimulala**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/42791

Problem Summary:

- Change the total retry time from 600ms to 2s.
-  Update logs.
- Add `MockDisableDistTask` to stop the dispatcher in the domain.

- Besides, clean up the cache in the test.

```
    dispatcher_test.go:212: 
        	Error Trace:	disttask/framework/dispatcher/dispatcher_test.go:21
        	            				disttask/framework/dispatcher/dispatcher_test.go:248
        	Error:      	Not equal: 
        	            	expected: 0
        	            	actual  : 1
        	Test:       	TestParallelNormalFlow
```

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: domain/domain.go [hunk 1]
```diff
@@ -1346,6 +1346,11 @@ func (do *Domain) checkReplicaRead(ctx context.Context, pdClient pd.Client) erro
 }
 
 func (do *Domain) initDistTaskLoop(ctx context.Context) error {
+	failpoint.Inject("MockDisableDistTask", func(val failpoint.Value) {
```
Comment (zimulala) [inline domain/domain.go:1349]: Add this to make `schedulerManager` can't be started.

**Review by hawkingrei — APPROVED** [review]

**Review by Defined2014 — APPROVED** [review]

**Pull request merged**